src/Swarm/Game/Exception.hs

```diff
@@ -43,18 +75,84 @@ data Exn
     User Text
   deriving (Eq, Show)
 
--- | Pretty-print an exception for displaying to the user.
-formatExn :: Exn -> Text
-formatExn (Fatal t) =
-  T.unlines
-    [ T.append "fatal error: " t
-    , "Please report this as a bug at https://github.com/swarm-game/swarm/issues/new ."
-    ]
-formatExn InfiniteLoop = "Infinite loop detected!"
-formatExn (Incapable _caps tm) =
-  T.concat
-    [ "missing device(s) needed to execute command "
-    , squote (prettyText tm)
-    ]
-formatExn (CmdFailed c t) = T.concat [prettyText c, ": ", t]
-formatExn (User t) = T.concat ["user exception: ", t]
+-- | Pretty-print an exception for displaying to the player.
+formatExn :: EntityMap -> Exn -> Text
+formatExn em = \case
+  Fatal t ->
+    T.unlines
+      [ "Fatal error: " <> t
+      , "Please report this as a bug at"
+      , "<https://github.com/swarm-game/swarm/issues/new>."
+      ]
+  InfiniteLoop -> "Infinite loop detected!"
+  (CmdFailed c t) -> T.concat [prettyText c, ": ", t]
+  (User t) -> "Player exception: " <> t
+  (Incapable f caps tm) -> formatIncapable em f caps tm
+
+-- ------------------------------------------------------------------
+-- INCAPABLE HELPERS
+-- ------------------------------------------------------------------
+
+formatIncapableFix :: IncapableFix -> Text
+formatIncapableFix = \case
+  FixByInstall -> "install"
+  FixByObtain -> "obtain"
+
+-- | Pretty print the incapable exception with an actionable suggestion
+--   on how to fix it.
+--
+-- >>> w = mkEntity (defaultEntityDisplay 'l') "magic wand" [] [] [CAppear]
+-- >>> r = mkEntity (defaultEntityDisplay 'o') "the one ring" [] [] [CAppear]
+-- >>> m = buildEntityMap [w,r]
+-- >>> incapableError cs t = putStr . unpack $ formatIncapable m FixByInstall cs t
+--
+-- >>> incapableError (S.singleton CGod) (TConst As)
+-- Can not perform an impossible task:
+--   'as'
+--
+-- >>> incapableError (S.singleton CAppear) (TConst Appear)
+-- You do not have the devices required for:
+--   'appear'
+--   please install:
+--    - the one ring or magic wand
+--
+-- >>> incapableError (S.singleton CCreate) (TConst Create)
+-- Missing the create capability for:
+--   'create'
+--   but no device yet provides it. See
+--   https://github.com/swarm-game/swarm/issues/26
+formatIncapable :: EntityMap -> IncapableFix -> Set Capability -> Term -> Text
+formatIncapable em f caps tm
+  | CGod `S.member` caps =
+    unlinesExText
+      [ "Can not perform an impossible task:"
```

Comment:
I think this error message could still be improved.  I don't think it is going to make much sense if a player stumbles across it.  *e.g.* imagine that they type `as` accidentally and notice that it has a type, then they wonder what it does so they try it out. They are going to wonder *why* the task is impossible. Somehow we should convey that this is a command that is not available during normal play.  We can definitely make it funny though.  Some ideas:
- "This command is not available to mere mortals" / "This command is forbidden"
- "Thou shalt not utter such blasphemy"
- If we wanted to be more informative and less funny, "This command is only available to system robots and scenario definitions".

Reply:
I guess the Elden ring is now popular, so blasphemy it is:
```Haskell
incapableError (S.singleton CGod) (TConst As)
-- Thee shalt not utter such blasphemy:
--   'as'
--   If't be true thee wanteth to playeth god, then tryeth Creative game.
```